### PR TITLE
Added configuration sourcing through web.config

### PR DESCRIPTION
**Pull request by cyberprune**
I have added sourcing the configuration through the web.config, this is useful for Azure Websites so you can put the configuration in the portal, rather than check the credentials into source control as part of the FileSystemProviders.config.

When no parameters are specified under the <Provider> node in FileSystemProviders.config, then the values are loaded from web.config.

### CODE REVIEW

File: src/UmbracoFileSystemProviders.Azure/AzureBlobFileSystem.cs [hunk 1]
```diff
@@ -12,13 +12,33 @@ namespace Our.Umbraco.FileSystemProviders.Azure
     using System.IO;
 
     using global::Umbraco.Core.IO;
```
Comment (JimBobSquarePants) [inline src/UmbracoFileSystemProviders.Azure/AzureBlobFileSystem.cs]: Umbraco?

**Comment by JimBobSquarePants** [comment]
Oh thanks for this! I'll give the code a good run through asap and get back to you. :+1:

**Comment by cyberprune** [comment]
Thanks, let me know if you spot something wrong, it is working nicely on my test project.

**Comment by JimBobSquarePants** [comment]
Hi @cyberprune Apologies for such a slow response. I'm just getting around to having a proper look at  everything again. Its been a very busy few months. 

Everything looks pretty good but I just have a question regarding your configuration setup? Are you configuring the provider as follows?

**FileSystemProviders.config**
``` xml
<?xml version="1.0"?>
<FileSystemProviders>
  <Provider alias="media" type="Our.Umbraco.FileSystemProviders.Azure.AzureBlobFileSystem, Our.Umbraco.FileSystemProviders.Azure">
    <Parameters>
    <!-- No Parameters: These are configured via appsettings in the Web.Config -->
    </Parameters>
  </Provider>
</FileSystemProviders>
```
**Web.config**
```xml
<?xml version="1.0"?>
<configuration>
  <appSettings>
    <add key="AzureBlobFileSystem.ConnectionString" value="DefaultEndpointsProtocol=https;AccountName=[myAccountName];AccountKey=[myAccountKey]" />
    <add key="AzureBlobFileSystem.ContainerName" value="media" />
    <add key="AzureBlobFileSystem.RootUrl" value="http://[myAccountName].blob.core.windows.net/" />
    <add key="AzureBlobFileSystem.MaxDays" value="365" />
  </appSettings>
</configuration>
```
If so, I'm happy to go ahead and add this as it makes sense.

Cheers

James

**Comment by JimBobSquarePants** [comment]
Actually scratch that, I was just reading @petervandenhout 's PR that builds on your one and I can see the configuration guide. I'll merge it in now.

**Pull request merged**